components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java

```diff
@@ -1274,12 +1274,16 @@ private boolean isDefaultGiven(String swaggerContent) throws APIManagementExcept
      */
     @Override
     public String processOtherSchemeScopes(String swaggerContent) throws APIManagementException {
-        if (!isDefaultGiven(swaggerContent)) {
-            Swagger swagger = getSwagger(swaggerContent);
-            swagger = processLegacyScopes(swagger);
+        Swagger swagger = getSwagger(swaggerContent);
+        Set<Scope> legacyScopes = getScopesFromExtensions(swagger);
+
+        if (!isDefaultGiven(swaggerContent) && legacyScopes.isEmpty()) {
             swagger = injectOtherScopesToDefaultScheme(swagger);
             swagger = injectOtherResourceScopesToDefaultScheme(swagger);
             return getSwaggerJsonString(swagger);
+        } else if (!legacyScopes.isEmpty()) {
```

Comment:
same goes here

Reply:
No. it is empty LinkedHashSet returning from getScopesFromExtensions